### PR TITLE
osv: canonicalize ranges

### PR DESCRIPTION
**Pull request by hdonnay**
This commit makes the OSV updater actually bother to fix non-canonical version ranges, instead of just complaining about them into the log.

### CODE REVIEW

**Review by crozzy — COMMENTED** [review]
I think we discussed throwing these out until we get reasonable data?

File: updater/osv/osv.go [hunk 1]
```diff
@@ -626,6 +627,34 @@ func (e *ecs) NewVulnerability() *claircore.Vulnerability {
 	return &e.Vulnerability[i]
 }
 
+// RemoveVulnerability does what it says on the tin.
+//
+// Will cause copying if the vulnerability is not the most recent returned from
+// NewVulnerability.
+func (e *ecs) RemoveVulnerability(v *claircore.Vulnerability) {
```
Comment (crozzy) [inline updater/osv/osv.go:634]: Does this need to be public?

**Review by crozzy — COMMENTED** [review]
LGTM, just one nit

File: updater/osv/osv.go (hunk 1 above)
Comment (hdonnay) [reply updater/osv/osv.go:634]: The receiver type is unexported, so this just keeps with the convention.

**Review by crozzy — APPROVED** [review]

**Pull request merged**